tasks/publish-to-mrrc/publish-to-mrrc.yaml

```diff
@@ -0,0 +1,75 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: publish-to-mrrc
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  params:
+    - name: mrrcParamFilePath
+      description: path of the env file for mrrc parameters to use
+      type: string
+    - name: cosignPubKeyConfig
+      description: the configmap name for cosign verify public key
+      type: string
+    - name: charonAWSSecret
+      description: the secret name for charon aws credential file
+      type: string
+  steps:
+    - name: verify-repo-signature
+      image: registry.redhat.io/rhtas/cosign-rhel9@sha256:75791f7ee9257e837d1e3db13415d858c397e803ffa4534c51ca19a2ba34b2ab
+      script: |
+        set -eux
+        MRRC_FILE="$(workspaces.data.path)/$(params.mrrcParamFilePath)"
+        . "$MRRC_FILE"
+        echo "Verify the maven repo zip $MRRC_ZIP_REGISTRY"
+        cosign verify --key /home/konflux/.cosign/cosign.pub $MRRC_ZIP_REGISTRY
+      volumeMounts:
+        - name: "pub-key-vol"
+          mountPath: "/home/konflux/.cosign/"
+    - name: download-maven-repo
+      image: quay.io/konflux-ci/release-service-utils:2a7815cc4bed65918422cf3c9fab4d2d3f298f58
+      script: |
+        set -eux
+        WORK_DIR="$(workspaces.data.path)"
+        MRRC_FILE="$WORK_DIR/$(params.mrrcParamFilePath)"
+        . "$MRRC_FILE"
+        echo "Downloading the maven repo zip $MRRC_ZIP_REGISTRY"
+        mrrc_dir="$(dirname $MRRC_FILE)/mrrc"
+        mkdir $mrrc_dir
+        cd $mrrc_dir
+        oras pull $MRRC_ZIP_REGISTRY
+    - name: upload-maven-repo
+      image: quay.io/redhat-services-prod/spmm-charon-containe-tenant/charon:e9312870607e2ba335231047af9ea81a03e52778
+      script: |
+        set -eux
+        MRRC_FILE="$(workspaces.data.path)/$(params.mrrcParamFilePath)"
+        . "$MRRC_FILE"
+        target=$MRRC_TARGET
+        productName=$MRRC_PRODUCT_NAME
+        productVersion=$MRRC_PRODUCT_VERSION
+        work_dir="$(dirname $MRRC_FILE)/mrrc"
+        mavenRepoPath="$(ls $work_dir/*.zip)"
+        echo "Release $productName-$productVersion into $target"
+        charon upload -p $productName -v $productVersion -t $target $mavenRepoPath
+      volumeMounts:
+        - name: "charon-configs-vol"
+          mountPath: "/home/charon/.charon"
+        - name: "charon-aws-vol"
+          mountPath: "/home/charon/.aws"
+  volumes:
+    - name: "pub-key-vol"
+      configMap:
+        name: "$(params.cosignPubKeyConfig)"
+    - name: "charon-configs-vol"
+      configMap:
+        name: "charon-configs"
```

Comment:
Have we thought about how this configMap is going to end up on the konflux clusters?

Reply:
I think this needs help from admins in rhtap-releng workspace to set it up. But I'm also considering if we should directly set it up in this task or in RPA as it's a yaml file in this configmap. Not sure which way is better from Konflux POV.

Reply:
> But I'm also considering if we should directly set it up in this task

You mean hardcode the settings in the task? I'd say the configMap sounds probably better, yeah. And if it's supposed to be the same for everybody, having it in RPA doesn't sound like a good fit. But I wonder what Johnny thinks.

Reply:
I am curious about this. Do the values just change for stage vs prod? Does something else make them change? We do things like this https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/push-disk-images/push-disk-images.yaml#L45 in other tasks, which sounds like it could work here. Would that work here? Avoiding the configMap would be nice because we need a way to deploy it on all of our clusters + we will have to deploy it on the ephemeral clusters that run e2e tests using this task

Reply:
A sample of this "charon-configs" is like this one: https://github.com/Commonjava/charon/blob/main/config/charon.yaml.sample
This can be changed by user, but if no exception in the final managed workspace, it will be a fixed content whatever the environment will be. But the content may be a little large, so I'm not sure if it's a good idea to hardcode it into task itself or a RPA.

Reply:
And another idea is storing this config file in the internal gitlab and give a url of it into RPA, then download it in this task. This will not need the configmap anymore. WDYT?

Reply:
I'm fine with having it in a configMap. I mean there is still no standardized way of managing configMaps and secrets that are needed by the release pipeline which is not great. But I heard that they will use RemoteSecrets in the future. Not sure about configMaps though. See also my comment here: https://github.com/konflux-ci/release-service-catalog/pull/510#discussion_r1745293693

Reply:
My issue with the configMap is what is creating it? We don't migrate clusters daily or something, but it does happen somewhat regularly. When we migrate, what if @ligangty is on PTO? Who creates this configMap? Do they know what to put in it? I don't really. Any manual step after provisioning a new cluster like creating a configMap is not too good

Reply:
The charon config yaml file will be changed in different environment(dev vs stage vs prod). I think the concerns with configmap make sense. I'll try to implement based on the RPA solution. But for the secret "awsSecret" here, I really don't know how to use some other ways except the k8s secret now in Konflux. Do you have some recommendations?

Reply:
We have different environments for lots of things, including secrets. Like here is an example https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/push-disk-images/push-disk-images.yaml#L46 and here https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/create-advisory/create-advisory.yaml#L84 you can see us checking the env by what image styles are provided. As far as secrets, https://redhat-internal.slack.com/archives/C031USXS2FJ/p1725539381988949 maybe

Reply:
> As far as secrets, https://redhat-internal.slack.com/archives/C031USXS2FJ/p1725539381988949 maybe

The dist of it is that in the future this will be managed via RemoteSecrets, but for now secrets are added manually by Scott or Ralph as needed.

So I believe for now the secret is fine the way you have it.

Reply:
@johnbieren @mmalina For the configmap issue, I'd like to do a separate PR later after this PR because it also needs to change the "collect-mrrc-params" together to accommodate the function. So can we close this thread in this PR?